Sources/Features/Breadcrumb/BacktraceBreadcrumbFile.swift

```diff
@@ -0,0 +1,119 @@
+import Foundation
+
+enum BacktraceBreadcrumbFileError: Error {
+    case invalidFormat
+}
+
+@objc class BacktraceBreadcrumbFile: NSObject {
+
+    private static let minimumQueueFileSizeBytes = 4096
+    private let maximumIndividualBreadcrumbSize: Int
+    private let maxQueueFileSizeBytes: Int
+    private let queue: Queue<Any>
+    private let breadcrumbLogURL: URL
+    private let dispatchQueue = DispatchQueue(label: "io.backtrace.BacktraceBreadcrumbFile@\(UUID().uuidString)")
+
+    public init(_ breadcrumbSettings: BacktraceBreadcrumbSettings) throws {
+        self.breadcrumbLogURL = try breadcrumbSettings.getBreadcrumbLogPath()
+        self.queue = Queue<Any>()
+        self.maximumIndividualBreadcrumbSize = breadcrumbSettings.maxIndividualBreadcrumbSizeBytes
+        if breadcrumbSettings.maxQueueFileSizeBytes < BacktraceBreadcrumbFile.minimumQueueFileSizeBytes {
+            BacktraceLogger.warning("\(breadcrumbSettings.maxQueueFileSizeBytes) is smaller than the minimum of " +
+                                    "\(BacktraceBreadcrumbFile.minimumQueueFileSizeBytes)" +
+                                    ", ignoring value and overriding with minimum.")
+            self.maxQueueFileSizeBytes = BacktraceBreadcrumbFile.minimumQueueFileSizeBytes
+        } else {
+            self.maxQueueFileSizeBytes = breadcrumbSettings.maxQueueFileSizeBytes
+        }
+
+        super.init()
+    }
+
+    func addBreadcrumb(_ breadcrumb: [String: Any]) -> Bool {
+        do {
+            // Serialize breadcrumb: [String: Any] into Data
```

Comment:
do we need to add a comment we're serializing an object? The line below the comment explains everything. I think it's better to leave comments when you have complex logic that is not obvious and requires documentation. Serializing JSON is too obvious to document